### PR TITLE
Protecting Singleton From Reflection And Serialization

### PR DESCRIPTION
**Pull request by Piyush-Blackbuck-121**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

File: guide/english/software-engineering/design-patterns/singleton/index.md [hunk 1]
```diff
@@ -145,8 +145,10 @@ This simple code is all there to implement a singleton design pattern in iOS usi
 
 An important thing to note is that `static` properties and methods are lazy initialize by default meaning that it will not be instantiated until it is being called, therefore it provides some optimization.
 
+Also, Basic implementation of singleton is breakable by reflection, serialization and multithreading. So it needs extra care while implementing this pattern depending on the usecase. Here is [goodread](https://medium.com/exploring-code/how-to-make-the-perfect-singleton-de6b951dfdb0) about Singleton.
```
Comment (thecodingaviator) [inline guide/english/software-engineering/design-patterns/singleton/index.md]:
```suggestion
Also, Basic implementation of singleton is breakable by reflection, serialization and multi-threading. So it needs extra care while implementing this pattern depending on the use-case. Here is a [goodread](https://medium.com/exploring-code/how-to-make-the-perfect-singleton-de6b951dfdb0) about Singleton.
```

**Review by thecodingaviator — CHANGES_REQUESTED** [review]
Check comments

**Comment by RandellDawson** [comment]
@thecodingaviator This one has gone stale.

**Review by thecodingaviator — COMMENTED** [review]
Hi,

As there has been no response from your end for a while, we are closing this PR.

Feel free to request this PR to be reopened, if you still wish to make the changes you originally added.  Also, remember to update your fork, so you have the latest version of master in case it has changed significantly.

Happy Coding! ✈️

**Pull request closed without merging**